### PR TITLE
Framework: Remove deprecations slated for 3.6 removal

### PR DESCRIPTION
**Pull request by aduth**
This PR removes deprecations scheduled to be removed in the upcoming 3.6.0 release.

- wp.editor.editorMediaUpload has been removed. Please use wp.editor.mediaUpload instead.
- wp.utils.getMimeTypesArray has been removed.
- wp.utils.mediaUpload has been removed. Please use wp.editor.mediaUpload instead.
- wp.utils.preloadImage has been removed.
- supports.wideAlign has been removed from the Block API. Please use supports.alignWide instead.
- wp.blocks.isSharedBlock has been removed. Use wp.blocks.isReusableBlock instead.
- fetchSharedBlocks action (core/editor) has been removed. Use fetchReusableBlocks instead.
- receiveSharedBlocks action (core/editor) has been removed. Use receiveReusableBlocks instead.
- saveSharedBlock action (core/editor) has been removed. Use saveReusableBlock instead.
- deleteSharedBlock action (core/editor) has been removed. Use deleteReusableBlock instead.
- updateSharedBlockTitle action (core/editor) has been removed. Use updateReusableBlockTitle instead.
- convertBlockToSaved action (core/editor) has been removed. Use convertBlockToReusable instead.
- getSharedBlock selector (core/editor) has been removed. Use getReusableBlock instead.
- isSavingSharedBlock selector (core/editor) has been removed. Use isSavingReusableBlock instead.
- isFetchingSharedBlock selector (core/editor) has been removed. Use isFetchingReusableBlock instead.
- getSharedBlocks selector (core/editor) has been removed. Use getReusableBlocks instead.

Source: https://wordpress.org/gutenberg/handbook/reference/deprecated/#3-6-0

### CODE REVIEW

**Comment by aduth** [comment]
cc @gziolo I employed new CHANGELOG workflows proposed in #8768 and it feels quite effective, also helps shine light on the impact of breaking changes on versioning.

File: packages/editor/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## 3.0.0 (Unreleased)
+
+- Breaking: The `wideAlign` block supports hook has been removed. Use `alignWide` instead.
```
Comment (gziolo) [inline packages/editor/CHANGELOG.md]: Shouldn’t it also be included in blocks? Sometimes those changes touch more than one package.

**Review by gziolo — COMMENTED** [review]
I did quick check and this proposal for updates to the changelog files is awesome. This would definitely make publishing much easier if we could make it a part of the workflow. We might want to group changes by type, I explained it in more detailed way in docs changes PR.

File: packages/editor/CHANGELOG.md (hunk 1 above)
Comment (aduth) [reply packages/editor/CHANGELOG.md]: Hmm, I agree it's relevant for someone implementing a block (interfacing with `blocks`), but the logic itself lives in `editor`, and as part of a general workflow I'd be conscious of removing the "make me think" part involved with considering documenting cascading effects.

Which is all to say: I don't think we should include it in `blocks`.

**Review by youknowriad — APPROVED** [review]

**Pull request merged**